src/backend/oneapi/fft.cpp

```diff
@@ -90,12 +101,35 @@ Array<Tc> fft_r2c(const Array<Tr> &in, const int rank) {
         ::oneapi::mkl::dft::descriptor<precision,
                                        ::oneapi::mkl::dft::domain::REAL>;
 
+    // this section defines reversed dimension decl and output
+    // strides: onemkl is row-major and arrayfire it column
+    // major. since the TRANSPOSE option on the onemkl descriptor is
+    // broken, we're forced to shuffle data like this
```

Comment:
```suggestion
```
This comment is no longer relevant. We can get rid of this after the latest changes.